example/centos/centos-9-aarch64-image-installer.yaml

```diff
@@ -0,0 +1,576 @@
+              source: device
+              target: /
+    - name: efiboot-tree
+      build: name:build
+      stages:
+        - type: org.osbuild.grub2.iso
+          options:
+            product:
+              name: CentOS Stream
+              version: 9-stream
+            kernel:
+              dir: /images/pxeboot
+              opts:
+                - inst.stage2=hd:LABEL=CentOS-Stream-9-BaseOS-aarch64
+                - inst.ks=hd:LABEL=CentOS-Stream-9-BaseOS-aarch64:/osbuild.ks
+            isolabel: CentOS-Stream-9-BaseOS-aarch64
+            architectures:
+              - AA64
+            vendor: centos
+    - name: os
+      build: name:build
+      stages:
+        - otk.external.osbuild-make-depsolve-dnf4-rpm-stage:
+            packageset: ${packages.os}
+            gpgkeys:
+              otk.include: "common/gpgkeys.yaml"
+        - type: org.osbuild.fix-bls
+          options: {}
+        - type: org.osbuild.locale
+          options:
+            language: C.UTF-8
+        - type: org.osbuild.timezone
+          options:
+            zone: America/New_York
+        - type: org.osbuild.sysconfig
+          options:
+            kernel:
+              update_default: true
+              default_kernel: kernel
+            network:
+              networking: true
+              no_zero_conf: true
+        - type: org.osbuild.selinux
+          options:
+            file_contexts: etc/selinux/targeted/contexts/files/file_contexts
+    - name: bootiso-tree
+      build: name:build
+      stages:
+        - type: org.osbuild.mkdir
+          options:
+            paths:
+              - path: /images
+              - path: /images/pxeboot
+        - type: org.osbuild.copy
+          inputs:
+            tree:
+              type: org.osbuild.tree
+              origin: org.osbuild.pipeline
+              references:
+                - name:anaconda-tree
+          options:
+            paths:
+              - from: input://tree/boot/vmlinuz-8-2.fk1.aarch64
+                to: tree:///images/pxeboot/vmlinuz
+              - from: input://tree/boot/initramfs-8-2.fk1.aarch64.img
+                to: tree:///images/pxeboot/initrd.img
+        - type: org.osbuild.squashfs
+          inputs:
+            tree:
+              type: org.osbuild.tree
+              origin: org.osbuild.pipeline
+              references:
+                - name:rootfs-image
+          options:
+            filename: images/install.img
+            compression:
+              method: xz
+              options:
+                bcj: arm
+        - type: org.osbuild.truncate
+          options:
+            filename: images/efiboot.img
+            size: ${size_efiboot}
+        - type: org.osbuild.mkfs.fat
+          options:
+            volid: c81236c2
+          devices:
+            device:
+              type: org.osbuild.loopback
+              options:
+                filename: images/efiboot.img
+                size: 40960
```

Comment:
We should probably put this value into a define next to `size_efiboot` too - it's `size_efiboot` in sectors so we should add a comment there and then ideally look if we can use the partition table external to generate all of this, it seems to be an extermly simple partition table as defined in https://github.com/osbuild/images/blob/v0.89.0/pkg/image/anaconda_tar_installer.go#L25 and with the defines generated with that we could use `otk-make-partition-stages` and below use `otk-partitions-mounts-devices`.

Given that we have no mechanism right now in images to changes these sizes we could of course just extract a define and add a comment how to calculate the right sizes by hand. But it would be nice to explore if we can do this the external (and maybe tweak/fix as needed). To me at least it feels a bit like starting with tech-debt if we ignore this and when later we copy the image definitions to create e.g. rhel10 and need to change sizes we will need to do calculations like this by hand.

Reply:
Amended the last commit to do this (made the variable name verbose ;)).